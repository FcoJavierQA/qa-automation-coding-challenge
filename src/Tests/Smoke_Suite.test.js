import {render, screen, fireEvent} from '@testing-library/react';
import App from '../App'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

describe('Smoke Tests', () => {

    test('Validate initial elements displayed', async () =>{
        const tree = renderer.create(<App/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Verify that entering a search term in the text input field and clicking the "Go" button triggers a search', async () =>{
        render(<App/>);
        const userInput = screen.getByRole("textbox", {id : "username"});
        const goButton = screen.getByRole("button", {type : "submit"});
        fireEvent.change(userInput, { target: { value: 'FcoJavierQA' }});
        fireEvent.click(goButton);
        expect(await screen.findByText("Found 2 Repos")).toBeInTheDocument();
        expect(await screen.findByText("ReportTest")).toBeInTheDocument();
    });

    test('Verify that if the error is due to a user not found on Github correct message is displayed', async () =>{
        render(<App/>);
        const userInput = screen.getByRole("textbox", {id : "username"});
        const goButton = screen.getByRole("button", {type : "submit"});
        fireEvent.change(userInput, { target: { value: 'Bad Information' }});
        fireEvent.click(goButton);
        expect(await screen.findByText("Github user not found")).toBeInTheDocument();
        expect(await screen.findByText("No repos")).toBeInTheDocument();
    });

});