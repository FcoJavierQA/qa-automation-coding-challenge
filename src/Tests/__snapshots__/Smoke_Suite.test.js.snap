// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke Tests Validate initial elements displayed 1`] = `
<div
  className="app"
  data-testid="main-app"
>
  <header>
    <h1>
      Get Github Repos
    </h1>
  </header>
  <main>
    <section
      className="message-area"
    />
    <form
      className="input-area"
    >
      <div
        className="field-username"
      >
        <label
          htmlFor="username"
        >
          Github Username
        </label>
        <input
          id="username"
          onChange={[Function]}
        />
      </div>
      <button
        className="submit"
        onClick={[Function]}
        type="submit"
      >
        Go
      </button>
    </form>
    <section
      className="output-area"
    >
      <p
        className="output-status-text"
      >
        No repos
      </p>
    </section>
  </main>
</div>
`;
